src/main/java/org/opentripplanner/middleware/connecteddataplatform/ConnectedDataManager.java

```diff
@@ -0,0 +1,334 @@
+                    Filters.eq("batchId", uniqueBatchId)
+                ),
+                Sorts.descending("dateCreated", "batchId")
+            );
+
+            TripRequest tripRequest = getTripRequestWithMaxModes(tripRequests);
+
+            // Get all trip summaries matching the batch id.
+            FindIterable<TripSummary> tripSummaries = Persistence.tripSummaries.getFiltered(
+                eq("batchId", uniqueBatchId),
+                Sorts.descending("dateCreated")
+            );
+
+            // Get place coordinates.
+            Coordinates fromCoordinates = getPlaceCoordinates(
+                tripSummaries,
+                true,
+                tripRequest.fromPlace,
+                isTest
+            );
+            Coordinates toCoordinates = getPlaceCoordinates(
+                tripSummaries,
+                false,
+                tripRequest.toPlace,
+                isTest
+            );
+
+            // Anonymize trip request.
+            AnonymizedTripRequest anonymizedTripRequest = new AnonymizedTripRequest(
+                tripRequest,
+                fromCoordinates,
+                toCoordinates
+            );
+
+            // Extract trip summaries and convert to anonymized trip summaries list.
+            List<AnonymizedTripSummary> anonymizedTripSummaries = tripSummaries
+                .map(tripSummary -> new AnonymizedTripSummary(tripSummary, fromCoordinates, toCoordinates))
+                .into(new ArrayList<>());
+
+            // Append content to file
+            FileUtils.writeToFile(
+                pathAndFileName,
+                true,
+                JsonUtils.toJson(new AnonymizedTrip(anonymizedTripRequest, anonymizedTripSummaries))
+            );
+            if (pos < numberOfUniqueBatchIds) {
+                // A comma is not required if processing the last item in the list. This is to prevent JSON formatting
+                // errors.
+                FileUtils.writeToFile(pathAndFileName, true, ",");
+            }
+        }
+        FileUtils.writeToFile(pathAndFileName, true, "]");
+    }
+
+    /**
+     * Workout if the first or last leg is public (a transit leg). If the leg is public the coordinates provided by OTP
+     * can be used. If not they are randomized. The place value is assumed to be in the format 'location :: lat,lon'.
+     */
+    private static Coordinates getPlaceCoordinates(
+        FindIterable<TripSummary> tripSummaries,
+        boolean isFirstLeg,
+        String place,
+        boolean isTest
+    ) {
+        boolean placeIsPublic = true;
+        for (TripSummary tripSummary : tripSummaries) {
+            placeIsPublic = isLegTransit(tripSummary.itineraries, isFirstLeg);
+            if (!placeIsPublic) {
+                // if a single trip summary is not public the place lat/lon must be randomized.
+                break;
+            }
+        }
+
+        String coords = place.split("::")[1].trim();
```

Comment:
Careful, the UI might send just the coordinates (if the geocoder does not return anything, which is unlikely).
If that happens, the format will just be `lat,lon` and `::` will not be present.